### DISCUSSION TITLE
How to extract only highlighted text?

### DISCUSSION THREAD

**Discussion by FRIKIdelTO** [jsvine/pdfplumber#1211] [Q&A]
I need to extract only highlighted text from a certain PDF file but I can't with this library.
I can do this with other libraries but I like to do it with this because I have better results.
Is there any way to do that?
Thanks

**Comment by jsvine** [comment]
Hi @FRIKIdelTO, and thanks for your interest in `pdfplumber`. In general, (digital) highlights are represented in PDFs via annotations. You can access annotations via `page.annots`. The specific representation of the annotation will likely differ based on the tool that created the highlights, but should contain useful positional information you can use in combination with `page.crop(...).extract_text()` to identify the highlighted text.

**Reply by FRIKIdelTO** [reply]
The box containing the highlighted text is extracted incorrectly on the Y-axis.
Code example to test it:
```
import pdfplumber
import fitz

pdf_path = "documento.pdf"

print('pdfplumber rects:')
with pdfplumber.open(pdf_path) as pdf:
    for page in pdf.pages:
        for annot in page.annots:
            if "Highlight" in str(annot["data"]["Subtype"]):
                print(annot["data"]["Rect"])

print()

print('PyMuPDF rects:')
with fitz.open(pdf_path) as pdf:
    for n in range(len(pdf)):
        page = pdf.load_page(n)
        for annot in page.annots():
            if annot.type[0] == 8:
                print(annot.rect)
```

example output:
```
pdfplumber rects:
[27.7573, 302.109, 784.988, 339.622]
[27.7573, 216.357, 784.988, 253.798]
[697.309, 436.54, 788.0, 462.285]

PyMuPDF rects:
Rect(27.757299423217773, 255.65399169921875, 784.9879760742188, 293.1669921875)
Rect(27.757299423217773, 341.4779968261719, 784.9879760742188, 378.91900634765625)
Rect(697.3090209960938, 132.99099731445312, 788.0, 158.73599243164062)
```

If I use PyMuPDF rects it works fine, but with pdfplumber rects extracts wrong text

**Reply by jsvine** [reply]
Thanks for the additional information and context. The values within the `["data"]` sub-dictionary are not modified from the raw values provided via `pdfminer.six`. Those unmodified positional values are based on a bottom-left 0,0-origin, rather than top-left (which is the origin used for `pdfplumber`'s chars, etc.). For positional information using the top-left origin, you'll want to check out the annotation's main positional keys: `x0`, `top`, `x1`, and `bottom`.
